### ISSUE TITLE
[Keyboard, Android] Modifier key_down can have 0 modifier, leading to crash (key 0x200000105, not SHIFT)

### ISSUE THREAD

**Issue by mikeesouth** [flutter/flutter#113592]
## Steps to Reproduce

1. Use a Chromebook with Android 11 (not sure how necessary this step is)
2. Focus a TextField
3. Press the Swedish letter 'ä' on the physical keyboard

**Expected results:** That the letter 'ä' is added to the text field text.

**Actual results:** Instant crash of application

## Flutter version
This can probably affect different versions of Flutter but I have this frequently reported from end users on Flutter 3.3.1 stable on Chromebooks with Android 11 with the CPU "Intel Jasper Lake".

## Related issues
This issue is practically the same as #108124 but the fix for that issue seems to be specific to the SHIFT key (0x200000102). My crash happens for the Swedish letter 'ä' and I'm not sure if the fix in #108124 only affects SHIFT or if it includes other keys too. The crash does not happen for the other two Swedish letters: 'å' and 'ö', these works as expected. I'm not sure what Flutter stable version includes the fix for #108124. Any tips on how I can see what Flutter version includes the fix would be greatly appreciated.

This crash is also very similar to #110653 but that is also shift key (0x200000102) where there is a video of the issue. That crash is exactly the same that I get but I get it with another key (0x200000105, swedish 'ä').

Another similar issue is #110640 that also addresses the shift key (0x200000102).

## Chrome OS version
I'm not sure if this is relevant but I do mention it here. We see this crash today (18th of October 2022) and it's 100% reproducible on the affected devices. I did test it on another Chromebook and we could reproduce the bug 100%. After updating that ChromeOS to 107.0.5083.0 (64 bit official build) the error went away on that device. I'm not sure which ChromeOS version the device had before, when the error was reproducible.
**To summarize:** The app is built on Flutter 3.3.1 stable and it crashed 100% of the time when we pressed 'ä'. Then I updated ChromeOS to 107.0.5083.0 and after that the crash cannot be reproduced anymore (I did not update the app).

<details>
<summary>Crash log (from Firebase Analytics)</summary>

``` 
Fatal Exception: java.lang.AssertionError: Unexpected metaState 0 for key 0x200000105 during an ACTION_down event.
       at io.flutter.embedding.android.KeyEmbedderResponder.synchronizePressingKey(KeyEmbedderResponder.java:127)
       at io.flutter.embedding.android.KeyEmbedderResponder.handleEventImpl(KeyEmbedderResponder.java:50)
       at io.flutter.embedding.android.KeyEmbedderResponder.handleEvent(KeyEmbedderResponder.java)
       at io.flutter.embedding.android.KeyboardManager.handleEvent(KeyboardManager.java:31)
       at io.flutter.embedding.android.FlutterView.dispatchKeyEvent(FlutterView.java:42)
       at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
       at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
       at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
       at com.android.internal.policy.DecorView.superDispatchKeyEvent(DecorView.java:476)
       at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1861)
       at android.app.Activity.dispatchKeyEvent(Activity.java:4095)
       at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:390)
       at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:5990)
       at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5858)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5353)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5410)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5376)
       at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5528)
       at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5384)
       at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5585)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5357)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5410)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5376)
       at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5384)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5357)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5410)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5376)
       at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5561)
       at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:5719)
       at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:3179)
       at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:2721)
       at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:2712)
       at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:3156)
       at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:143)
       at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
       at android.os.MessageQueue.next(MessageQueue.java:335)
       at android.os.Looper.loop(Looper.java:183)
       at android.app.ActivityThread.main(ActivityThread.java:7737)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
```
</details>

**Comment by huycozy** [comment]
Hi @mikeesouth 
As far as I can see, the patch fix for #110640 is available from `3.3.3` onwards, please see https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#333-sept-28-2022.

So, please try to upgrade to the latest Flutter stable channel (`3.3.4`) and retry.

**Comment by mikeesouth** [comment]
@huycozy thanks, I missed that issue number on the hotfixes page! I will try a new build with 3.3.4 (it's already on the way to end users). I will update with results here later this week.
